include/envoy/stream_info/stream_info.h

```diff
@@ -78,8 +78,10 @@ enum ResponseFlag {
   ResponseFromCacheFilter = 0x100000,
   // Filter config was not received within the permitted warming deadline.
   NoFilterConfigFound = 0x200000,
+  // Request or connection exceeded the downstream connection duration.
+  DurationTimeout = 0x400000,
```

Comment:
When you hit a `max_connection_duration` with HCM and downstream at L7, what response flag is produced today?

Reply:
Will be same as before.I have set the response flag for tcp connection only as of now.

Reply:
My question is what was the flag before, and why do we have different behavior for TCP and HTTP connection-level timeouts in the flags?

Reply:
absl::nullopt is set as reponse flag.

Reply:
Makes sense. If you don't want to update HCM in this PR it's fine, but maybe add a TODO there and file a tracking issue.

Reply:
Done.